src/bloaty.cc

```diff
@@ -914,24 +920,61 @@ class MmapInputFile : public InputFile {
   BLOATY_DISALLOW_COPY_AND_ASSIGN(MmapInputFile);
 };
 
+#if defined(_MSC_VER)
```

Comment:
Instead of `#ifdef`ing inside `FileDescriptor` and `MmapInputFile`, I'd prefer to use separate classes. `InputFile` is a virtual base, so we can create a different derived class `WindowsInputFile`.

Generally I prefer to keep `#ifdef` to a minimum.

Reply:
In that case I would have to ifdef out the `MmapInputFile` and `FileDescriptor` entirely on windows (I'll use `_MSC_VER`, unless you have a preference for `_WIN32` instead?)

Reply:
Yes, I think that's a good approach. Later we could split off the UNIX or Windows-specific code into separate files like `unix.cc` and `windows.cc`, especially if the amount of platform-specific code grows, but I don't think that's necessary now.